spec/lib/rbac/filterer_spec.rb

```diff
@@ -211,15 +211,55 @@ def combine_filtered_ids(user_filtered_ids, belongsto_filtered_ids, managed_filt
     end
 
     context 'when class does not participate in RBAC' do
+      before do
+        @vm       = FactoryGirl.create(:vm_vmware,       :name => "VM1",       :host => @host1, :ext_management_system => @ems)
+
+        @timestamps = [
+            ["2010-04-14T20:52:30Z", 100.0],
+            ["2010-04-14T21:51:10Z", 1.0]
+        ]
+        @timestamps.each do |t, v|
+          @vm.metric_rollups << FactoryGirl.create(:metric_rollup_vm_hr,
+                                                   :timestamp                  => t,
+                                                   :cpu_usage_rate_average     => v,
+                                                   :cpu_ready_delta_summation  => v * 1000, # Multiply by a factor of 1000 to maake it more realistic and enable testing virtual col v_pct_cpu_ready_delta_summation
+                                                   :sys_uptime_absolute_latest => v
+          )
+        end
+      end
       let(:miq_ae_domain) { FactoryGirl.create(:miq_ae_domain) }
 
-      it 'returns same class as input' do
+      it 'returns the same class as input for MiqAeDomain' do
         User.with_user(admin_user) do
           results = described_class.search(:targets => [miq_ae_domain]).first
           expect(results.first).to be_an_instance_of(MiqAeDomain)
           expect(results).to match_array [miq_ae_domain]
         end
       end
+
+      it 'returns the same class as input for parents clast that is not STI' do
+        User.with_user(admin_user) do
+          targets = @vm.metric_rollups
+
+          results = described_class.search(:targets => targets, :user => admin_user)
+          objects = results.first
+          expect(objects.length).to eq(2)
+          expect(objects).to match_array(targets)
+        end
+      end
+
+      it 'returns the same class as input for subclass that is not STI' do
```

Comment:
^ what does that mean?  Is it returning a base class or subclass?

Reply:
It's returning the subclass. But, the point is that it's returning the same class that was passed to targets.